### ISSUE TITLE
Unable to "continue" when we have a "Answer a question - Date/Time" bug

### ISSUE THREAD

**Issue by os-micmec** [google/ground-platform#1304]
**Describe the bug**
I am not able to "continue" on the "add tasks to job" step in the survey creation flow, when I have a "Answer a question - Date/Time" task. The error I see in the console is: 

```
core.mjs:8400 ERROR Error: Uncaught (in promise): Error: Unsupported task type 7
Error: Unsupported task type 7
    at FirebaseDataConverter.taskTypeToString (firebase-data-converter.ts:340:13)
    at FirebaseDataConverter.taskToJS (firebase-data-converter.ts:266:37)
    at firebase-data-converter.ts:171:36
    at immutable.es.js:2461:29
    at immutable.es.js:2479:16
    at ArrayMapNode.iterate.HashCollisionNode.iterate (immutable.es.js:2868:11)
    at Map.__iterate (immutable.es.js:2477:18)
    at Map.forEach (immutable.es.js:4854:17)
    at immutable.es.js:2460:11
    at Map.withMutations (immutable.es.js:2349:3)
    at resolvePromise (zone.js:1265:35)
    at zone.js:1172:21
    at zone.js:1188:37
    at asyncGeneratorStep (asyncToGenerator.js:6:1)
    at _throw (asyncToGenerator.js:25:1)
    at _ZoneDelegate.invoke (zone.js:412:30)
    at Object.onInvoke (core.mjs:24210:33)
    at _ZoneDelegate.invoke (zone.js:411:56)
    at Zone.run (zone.js:169:47)
    at zone.js:1329:38
```

**To Reproduce**
Create a new survey, and try to add (alone and/or with other tasks) a "Answer a question - Date/Time" task. Try to click continue, you should be stuck on the page. Now remove that task, you should be able to continue.

**Expected behavior**
Simple continue

**Comment by gino-m** [comment]
Other errors also precede the above trace:

```
Error: Ignoring unsupported task of type: point
    at FirebaseDataConverter.stringToTaskType (option.model.ts:1:1)
    at FirebaseDataConverter.toTask (survey.model.ts:39:14)
    at submission.model.ts:29:21
    at Array.map (<anonymous>)
    at FirebaseDataConverter.toTasks (submission.model.ts:29:21)
    at FirebaseDataConverter.toJob (result.model.ts:21:1)
    at role.model.ts:21:17
    at Array.map (<anonymous>)
    at FirebaseDataConverter.toSurvey (role.model.ts:21:17)
    at ground-pin.service.ts:23:21
toTask @ survey.model.ts:44
```

```
survey.model.ts:44 Error: Ignoring unsupported task of type: number
    at FirebaseDataConverter.stringToTaskType (option.model.ts:1:1)
    at FirebaseDataConverter.toTask (survey.model.ts:39:14)
    at submission.model.ts:29:21
    at Array.map (<anonymous>)
    at FirebaseDataConverter.toTasks (submission.model.ts:29:21)
    at FirebaseDataConverter.toJob (result.model.ts:21:1)
    at role.model.ts:21:17
    at Array.map (<anonymous>)
    at FirebaseDataConverter.toSurvey (role.model.ts:21:17)
    at ground-pin.service.ts:23:21
```

**Comment by rfontanarosa** [comment]
fixed by #1331
